docs/src/sections/solution/webapp.adoc

```diff
@@ -0,0 +1,480 @@
+
+* common - komponenty wykorzystywane w więcej niż jednych widoku
+* Context - komponenty zwracające _Reactowy context_ z przygotowanymi danymi i hookami
+* Form - komponenty do tworzenia formularzy
+* Layouts - komponenty, które wyświetlają inne komponenty dodając do nich pewne elementy
+* Modals - okna dialogowe
+
+Reszta katalogów w folderze _components_ jest nazwana od widoków, gdzie są wykorzystywane
+
+* auth
+* adminPanel
+* wallet
+* History
+* Quickbuy
+* Transactions
+
+==== Pliki konfiguracyjne
+
+* postcss.config.js - konfiguracja narzędzia _postcss_, wymagana przez _tailwindcss_
+* jest.config.js - zawiera konfiguracje takich rzeczy jak generowanie raportów z testów, czy dodawanie ścieżek modułów wykorzystywanych w testach
+* .prettierrc - konfiguracja narzędzia _prettier_
+* .tailwind.config.js - konfiguracja narzędzia _tailwind_
+* .vite.config.ts - zawiera ścieżki, których zapytania powinny być przekazane przez _vite_ do serwera, ponieważ w środowisku deweloperskim serwer i aplikacja webowa są uruchomione na różnych portach
+[source,typescript]
+----
+include::../../../../webapp/vite.config.ts[lines=7..20]
+----
+** .eslintrc - zawiera informacje o tym jakiego typu problemy ze składnią powinien zgłaszać _eslint_ przy analizowaniu kodu
+
+
+==== Nawigacja
+
+Do nawigacji została wykorzystana biblioteka _react-router-dom_.
+Przypisanie komponentów do ścieżek można zobaczyć w komponencie głównym aplikacji - _src/components/App.tsx_.
+
+.App.tsx
+[source, typescriptjsx]
+----
+include::../../../../webapp/src/components/App.tsx[lines=40..57]
+...
+----
+
+Ścieżki URL aplikacji webowej zostały zapisane w postaci _JSON_ do pliku _src/routesBuilder.ts_ w celu zwiększania elastyczności kodu.
+
+Oprócz zdefiniowanych ścieżek, widać zagnieżdżenie większości komponentów aplikacji w _contextach_ i _layoutach_.
+Layout _NavbarLayout.tsx_ zgnieżdża prawie każdy widok, ponieważ dodaje górny pasek nawigacyjny do widoków.
+Jedyne widoki, które owego paska nie posiadają są _LogoutPage.tsx_ i _Error404Page.tsx_.
+Do tego Sprawdza, czy _UserContext.tsx_ wczytuje token, jeśli tak to dodaje notyfikacje z informacją, że token jest w trakcie ładowania.
+
+.NavbarLayout.tsx
+[source, typescriptjsx]
+----
+include::../../../../webapp/src/components/Layouts/NavbarLayout.tsx[lines=9..32]
+----
+
+.Wiadomość o wczytywaniu tokenu w lewym dolnym rogu
+image::../../images/system_presentation/webapp/refreshing_token.png[]
+
+_UserLayout.tsx_, tak samo jak _AdminLayout.tsx_ i _PublicLayout.tsx_ przekierowuje użytkownika w zależności od jego stanu.
+Zagnieżdżają w sobie komponenty w zależności od tego komu należy się udostępnić dany widok.
+
+* _UserLayout.tsx_ - przekierowuje do strony logowania jeśli użytkownik nie jest zalogowany, lub do strony głównej jeśli użytkownik ma rolę tymczasową
+* _AdminLayout.tsx_ - przekierowuje do strony logowania jeśli użytkownik nie jest zalogowany, lub do strony głównej jeśli użytkownik nie jest adminem
```

Comment:
trochę ta linijka po powtórzenie poprzedniej.

Reply:
to zamierzany zabieg